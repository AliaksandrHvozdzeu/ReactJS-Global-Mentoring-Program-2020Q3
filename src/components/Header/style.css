header {
    height: 50vh;
    background-image: url(/images/header.jpg);
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    position: relative;
    overflow: hidden;
}

header::before {
    background-color: #00000096;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    content: '';
    backdrop-filter: blur(1px);
    z-index: 1;
}

header > * {
    position: relative;
    z-index: 2;
}
