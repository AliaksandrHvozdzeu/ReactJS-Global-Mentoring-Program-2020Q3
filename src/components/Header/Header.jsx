import React from 'react';
import PropTypes from 'prop-types';
import Search from '../SearchPanel';
import MoviePreview from '../MoviePreview';
import './style.css';

const Header = ({ details, closePreview, onFilterByName, filterByName }) => (
  <header id="header">
    {details && <MoviePreview details={details} closePreview={closePreview}/>}
    {!details && <Search filterByName={filterByName} onFilterByName={onFilterByName}/>}
  </header>
);

Header.propTypes = {
  details: PropTypes.string.isRequired,
  closePreview: PropTypes.func.isRequired,
  onFilterByName: PropTypes.string.isRequired,
  filterByName: PropTypes.string.isRequired
};

export default Header;
