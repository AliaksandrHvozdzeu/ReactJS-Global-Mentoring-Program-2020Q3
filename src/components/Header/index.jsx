import React from 'react';
import Search from '../SearchPanel';
import './style.css';
import MoviePreview from '../MoviePreview';

const Header = ({ details, closePreview, onFilterByName, filterByName }) => (
  <header id="header">
    {details && <MoviePreview details={details} closePreview={closePreview}/>}
    {!details && <Search filterByName={filterByName} onFilterByName={onFilterByName}/>}
  </header>
);

export default Header;
