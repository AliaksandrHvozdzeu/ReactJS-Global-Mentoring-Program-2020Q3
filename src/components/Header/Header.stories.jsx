import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import Header from './js/Header.container';
import '../../App.css';

export default {
  title: 'Header',
  decorators: [withKnobs],
};

export const Default = () => (
  <Header />
);
