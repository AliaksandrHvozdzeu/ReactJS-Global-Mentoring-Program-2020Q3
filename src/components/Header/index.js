import Header from './js/Header.container';

export default Header;
