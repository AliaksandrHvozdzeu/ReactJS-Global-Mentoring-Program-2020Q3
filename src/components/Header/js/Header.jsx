import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import SearchPanel from '../../SearchPanel';
// eslint-disable-next-line import/no-cycle
import MoviePreview from '../../MoviePreview';
import store from '../../../store';
import '../css/Header.css';

export default function Header({ details, blur }) {

  const buildHeader = () => {
    return details ? <MoviePreview details={details}/> : <SearchPanel/>;
  };

  return (
    <>
      <Provider store={store}>
        <header id="header" className={blur ? 'blurred' : ''}>
          {buildHeader()}
        </header>
      </Provider>
    </>
  );

}

Header.propTypes = {
  details: PropTypes.string.isRequired,
  blur: PropTypes.bool.isRequired,
};
