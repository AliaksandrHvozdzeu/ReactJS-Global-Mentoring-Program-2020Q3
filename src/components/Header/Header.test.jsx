import React from 'react';
import rerender from 'react-test-renderer';
import Header from './js/Header';
import { jest } from "@jest/globals";

describe('when Header', () => {

  test('then snapshot created when blurred', () => {
    const onBuildHeader = jest.fn();
    const component = rerender.create(<Header blur={true} onBuildHeader={onBuildHeader}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('then snapshot created when not blurred', () => {
    const onBuildHeader = jest.fn();
    const component = rerender.create(<Header blur={false} onBuildHeader={onBuildHeader}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
