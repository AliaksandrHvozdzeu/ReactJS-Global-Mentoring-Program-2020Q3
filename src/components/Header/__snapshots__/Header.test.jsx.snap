// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Header then snapshot created when blurred 1`] = `
<header
  className="blurred"
  id="header"
/>
`;

exports[`when Header then snapshot created when not blurred 1`] = `
<header
  className=""
  id="header"
/>
`;
